courses/tasks.py

```diff
@@ -0,0 +1,62 @@
+"""
+Tasks for the courses app
+"""
+import logging
+from mitxpro.celery import app
+from mitxpro.utils import now_in_utc
+from courses.models import CourseRun, CourseRunCertificate
+from courses.utils import ensure_course_run_grade, process_course_run_grade_certificate
+from courseware.api import get_edx_grades_with_users
+
+log = logging.getLogger(__name__)
+
+
+@app.task
+def generate_course_certificates():
+    """
+    Task to generate certificates for courses.
+    """
+    now = now_in_utc()
+    course_runs = CourseRun.objects.filter(end_date__lt=now).exclude(
+        id__in=CourseRunCertificate.objects.values_list("course_run__id", flat=True)
+    )
+
+    for run in course_runs:
+        edx_grade_user_iter = get_edx_grades_with_users(run)
```

Comment:
This would raise an exception if the corresponding course with `courseware_id` for whatever reason does not exist on Opn edX, effectively breaking the command. Might want to catch this exception and log it and continue.

Reply:
Then I believe, It should also be handled in previous [PR](https://github.com/mitodl/mitxpro/pull/1032/files#diff-d9b03b7190f4500a09f0487ac07ea510R69) of management command well too.